### PR TITLE
Adding duration information of the pipeline steps on Bears spreadsheet and also putting it back on Repairnator spreadsheet (issue #146)

### PR DESCRIPTION
**Pull request by fermadeiral**

### CODE REVIEW

**Comment by surli** [comment]
I propose to wait before merging this one: once #150 is merged, then you'll be able to add runid also in the other spreadsheet. WDYT?

**Comment by fermadeiral** [comment]
Ok. Until there, I will work in this same branch then and I will add `GoogleSpreadSheetInspectorTrackTreatedBuilds` and `GoogleSpreadSheetEndProcessSerializer` back. I'm not sure, but I think that `GoogleSpreadSheetEndProcessSerializer` should be moved to repairnator-pipeline. I will propose something in this direction and then we discuss if you don't agree.

**Comment by surli** [comment]
> but I think that GoogleSpreadSheetEndProcessSerializer should be moved to repairnator-pipeline.

I don't agree.

File: repairnator/repairnator-core/src/main/java/fr/inria/spirals/repairnator/serializer/GoogleSpreadSheetUtils.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.google.api.services.sheets.v4.model.ValueRange;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Created by fernanda on 22/03/17.
+ */
+public class GoogleSpreadSheetUtils {
```
Comment (surli) [inline repairnator/repairnator-core/src/main/java/fr/inria/spirals/repairnator/serializer/GoogleSpreadSheetUtils.java]: hmmm instead of creating a new class, can't you put that in GoogleSpreadsheetFactory?

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java [hunk 2]
```diff
@@ -218,12 +212,15 @@ private void mainProcess() throws IOException {
...
             serializers.add(new GoogleSpreadSheetInspectorSerializer(this.config.getGoogleSecretPath()));
+            serializers.add(new GoogleSpreadSheetInspectorTimeSerializer(this.config.getGoogleSecretPath()));
             serializers.add(new GoogleSpreadSheetNopolSerializer(this.config.getGoogleSecretPath()));
         } else {
             GoogleSpreadSheetFactory.setSpreadsheetId(GoogleSpreadSheetFactory.BEAR_SPREADSHEET_ID);
 
             serializers.add(new GoogleSpreadSheetInspectorSerializer4Bears(this.config.getGoogleSecretPath()));
+            serializers.add(new GoogleSpreadSheetInspectorTimeSerializer4Bears(this.config.getGoogleSecretPath()));
         }
+        serializers.add(new GoogleSpreadSheetInspectorTrackTreatedBuilds(this.buildToBeInspected, this.config.getGoogleSecretPath()));
```
Comment (surli) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java]: So we're agree that it should be move to dockerpool?

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/ProjectInspector4Bears.java [hunk 3]
```diff
@@ -35,33 +35,33 @@ public void run() {
...
+                    .setNextStep(new BuildProject(this, BuildProject.class.getSimpleName()+"Build"))
+                    .setNextStep(new TestProject(this, TestProject.class.getSimpleName()+"Build"))
+                    .setNextStep(new GatherTestInformation(this, new BuildShouldPass(), GatherTestInformation.class.getSimpleName()+"Build"))
                     .setNextStep(new CheckoutPreviousBuild(this))
-                    .setNextStep(new BuildProject(this))
-                    .setNextStep(new TestProject(this))
+                    .setNextStep(new BuildProject(this, BuildProject.class.getSimpleName()+"PreviousBuild"))
+                    .setNextStep(new TestProject(this, TestProject.class.getSimpleName()+"PreviousBuild"))
                     .setNextStep(gatherTestInformation)
                     .setNextStep(new PushIncriminatedBuild(this));
```
Comment (surli) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/ProjectInspector4Bears.java]: This is not up-to-date with last changes with squash step before push

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java (hunk 2 above)
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java]: Yes, we agree. This is only temporary.

File: repairnator/repairnator-core/src/main/java/fr/inria/spirals/repairnator/serializer/GoogleSpreadSheetUtils.java (hunk 1 above)
Comment (fermadeiral) [inline repairnator/repairnator-core/src/main/java/fr/inria/spirals/repairnator/serializer/GoogleSpreadSheetUtils.java]: Ok.

Comment (fermadeiral) [inline repairnator/repairnator-core/src/main/java/fr/inria/spirals/repairnator/serializer/GoogleSpreadSheetUtils.java]: Done.

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java (hunk 2 above)
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java]: This will be fixed in another pull request.

Comment (surli) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java]: Then could you comment that line, to be sure it's not used?

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/ProjectInspector4Bears.java (hunk 3 above)
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/ProjectInspector4Bears.java]: I will rebase and solve this.

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java (hunk 2 above)
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java]: Ok.

**Comment by fermadeiral** [comment]
Simon, can I update the `GoogleSpreadSheetInspectorTimeSerializer` for the other steps we added? This is about the "Duration Data" tab in the LibRepair spreadsheet.

**Comment by surli** [comment]
> Simon, can I update the GoogleSpreadSheetInspectorTimeSerializer for the other steps we added? This is about the "Duration Data" tab in the LibRepair spreadsheet.

ok but update the spreadsheet too then.

**Comment by surli** [comment]
@fermadeiral did you already try the code for serializers?

**Comment by fermadeiral** [comment]
> @fermadeiral did you already try the code for serializers?

The changes I made yesterday, yes. The changes I made today, no. I will try to do that now.

**Comment by surli** [comment]
>  I will try to do that now.

Tell me when it's done
